### ISSUE TITLE
StageTextTextEditor: support ability to remain in focus when touching another component

### ISSUE THREAD

**Issue by joshtynjala** [BowlerHatLLC/feathers#1200]
This is often requested for chat interfaces built with Feathers. Currently, if you tap a button to send a message, the TextInput loses focus, and it looks terrible. The problem is that the runtime does not expose an appropriate API to allow it.

If StageText could dispatch a cancelable FocusEvent.MOUSE_FOCUS_CHANGE, like other native Flash display objects, that would do the trick. I could call preventDefault() on the event, and the StageText would remain in focus.

I submitted a feature request to Adobe. Please vote!

[Adobe Feature Request #4057869: StageText: add FocusEvent.MOUSE_FOCUS_CHANGE event that allows preventDefault() to cancel focus loss ](https://bugbase.adobe.com/index.cfm?event=bug&id=4057869)

**Comment by joshtynjala** [comment]
Forgot that someone had opened this issue already:

[Adobe Feature #3797220: Soft keyboard maintain focus ](https://bugbase.adobe.com/index.cfm?event=bug&id=3797220)

Same idea, but mine includes a specific suggestion that matches behavior with other parts of AIR.

**Comment by khoamoc** [comment]
it seems like it was fixed. How do I use this in Feathers 3 text input?

**Comment by joshtynjala** [comment]
It is not exposed by StageTextTextEditor yet. If you want to use it now, you need to subclass and handle the StageText event manually.

**Comment by joshtynjala** [comment]
I'll try to take a look for 3.1.

**Comment by joshtynjala** [comment]
I have added a maintainTouchFocus property to StageTextTextEditor that uses the new FocusEvent.MOUSE_FOCUS_CHANGE event on StageText. On Android and iOS, focus is maintained when touching something else, like a button. On Mac, it has no effect, for some reason. The listener is called, but preventDefault() does nothing.

I tried adding a similar property on TextFieldTextEditor. However, it appears that calling preventDefault() in a FocusEvent.MOUSE_FOCUS_CHANGE listener no longer works for TextField. This was working in the past, so something seems to have broken.
